src/yetibot/commands/wolfram.clj

```diff
@@ -2,31 +2,63 @@
   (:require
     [clojure.spec.alpha :as s]
     [yetibot.core.util.http :refer [encode]]
-    [clojure.xml :as xml]
-    [taoensso.timbre :refer [info warn error]]
     [yetibot.core.config :refer [get-config]]
-    [yetibot.core.hooks :refer [cmd-hook]]))
+    [yetibot.core.hooks :refer [cmd-hook]]
+    [clojure.data.json :as json]
+    [taoensso.timbre :refer [error]]
+    [clojure.string :as string]))
 
 (s/def ::appid string?)
 
 (s/def ::config (s/keys :req-un [::appid]))
 
 (def config (:value (get-config ::config [:wolfram])))
 (def app-id (:appid config))
-(def endpoint (str "http://api.wolframalpha.com/v2/query?appid=" app-id))
+(def endpoint (str "http://api.wolframalpha.com/v2/query?output=json&appid="
+                   app-id))
 
-(defn parse-imgs-from-xml [xml]
-  (let [xs (xml-seq xml)]
-    (for [el xs :when (= :img (:tag el))] [(:alt (:attrs el)) (:src (:attrs el))])))
+(defn plaintext-or-image
+  "Return the subpods plaintext if present, otherwise it's image link."
+  [subpod]
+  (let [plaintext (:plaintext subpod)]
+    (if-not (empty? plaintext)
+      plaintext
+      (get-in subpod [:img :src]))))
+
+(defn data-string
+  "Concatenate pod title and all subpods' data."
+  [pod]
+  (->> pod
+       (:subpods)
+       (map plaintext-or-image)
+       (string/join "; ")
+       (str (:title pod) ": ")))
+
+(defn catch-suggestion
+  "Print some help if Wolfram can't interpret the query."
+  [wolfram-data]
+  (or (get-in wolfram-data [:tips :text])
+      (if-let [suggestion (get-in wolfram-data [:didyoumeans :val])]
+        (str "Did you mean " suggestion "?"))
+      (get-in wolfram-data [:languagemsg :english])
+      "Wolfram couldn't interpret your query."))
 
 (defn search-wolfram
   "wolfram <query> # search for <query> on Wolfram Alpha"
   {:yb/cat #{:info :img}}
   [{q :match}]
-  (flatten
-    (map #(str (second %) "&t=.jpg")
-         (parse-imgs-from-xml
-           (xml/parse (str endpoint "&input=" (encode q)))))))
+  (let [data (:queryresult (json/read-str
+                            (slurp (str endpoint "&input=" (encode q)))
```

Comment:
I realize the previous code was not great in terms of dynamically building a url. If you're looking to improve this, I'd probably switch to [clj-http](https://github.com/dakrone/clj-http) to clean up the http request. Something like:

```clojure
(client/get endpoint {:query-params {:input q :output "json" :appid app-id}
                      :as :json})
```

This handles query string encoding and json parsing of the response.